src/Symfony/Component/Translation/Translator.php

```diff
@@ -422,11 +420,9 @@ protected function computeFallbackLocales($locale)
     /**
      * Asserts that the locale is valid, throws an Exception if not.
      *
-     * @param string $locale Locale to tests
-     *
      * @throws InvalidArgumentException If the locale contains invalid characters
      */
-    protected function assertValidLocale($locale)
+    protected function assertValidLocale(?string $locale)
     {
         if (1 !== preg_match('/^[a-z0-9@_\\.\\-]*$/i', $locale)) {
```

Comment:
preg_match should not be a called with null.

Reply:
Also I assume the nullable locale comes from `setLocale`. But according to https://github.com/symfony/symfony/blob/4.4/src/Symfony/Contracts/Translation/LocaleAwareInterface.php it's not nullable. Seems like we need to fix the interface then.

Reply:
let's take this as an opportunity to remove the `1 !==` comparison, we don't do this in the codebase when using preg_*

Reply:
The translator contracts have no type declarations. Have those been skipped on purpose?

Reply:
The contracts need to support php 7.1. Adding parameter type declarations there would be a breaking change.

Reply:
The unit tests explicitly list `null` as a valid value for `$locale`: https://github.com/symfony/symfony/blob/ef0119a2d40f1a5930b4fd9146b1569b17d849ea/src/Symfony/Component/Translation/Tests/TranslatorTest.php#L489

Reply:
#33247